services/wallet/transfer/transaction_manager_multitransaction.go

```diff
@@ -39,8 +39,12 @@ func (tm *TransactionManager) CreateMultiTransactionFromCommand(command *MultiTr
 
 	multiTransaction := multiTransactionFromCommand(command)
 
-	if multiTransaction.Type == MultiTransactionSend && multiTransaction.FromNetworkID == 0 && len(data) == 1 {
-		multiTransaction.FromNetworkID = data[0].ChainID
+	// Set network for single chain transactions
+	switch multiTransaction.Type {
+	case MultiTransactionSend, MultiTransactionApprove, MultiTransactionSwap:
+		if multiTransaction.FromNetworkID == 0 && len(data) == 1 {
```

Comment:
```suggestion
		if multiTransaction.FromNetworkID == wallet_common.EthereumMainnet && len(data) == 1 {
```

Reply:
0 is the unknown chainID (ethMainnet is 1). Changed it to use the constant.

Reply:
exactly, not sure why I set Ethereum mainnet, but you got the point :)